astropy/time/core.py

```diff
@@ -1355,6 +1364,68 @@ def sort(self, axis=-1):
         return self[self._advanced_index(self.argsort(axis), axis,
                                          keepdims=True)]
 
+    def mean(
+            self: Self,
+            axis: int | tuple[int, ...] | None = None,
+            dtype: npt.DTypeLike = None,
+            out: Time | None = None,
+            keepdims: bool = False,
+            *,
+            where: npt.NDArray[bool] = True,
+    ) -> Self:
+        """Mean along a given axis.
+
+        This is similar to :meth:`~numpy.ndarray.mean`, but adapted to ensure
+        that the full precision given by the two doubles ``jd1`` and ``jd2``
+        is used, and that corresponding attributes are copied.
+
+        Note that the ``out`` argument is present only for compatibility with
+        ``np.mean``; since `Time` instances are immutable, it is not possible
+        to have an actual ``out`` to store the result in.
+
+        Similarly, the ``dtype`` argument is also present for compatibility with :meth:`~numpy.ndarray.mean`
```

Comment:
Total nitpick, but since docstrings can be shown on the terminal, maybe wrap this part to be <~80 chars/line